### PR TITLE
Change install_nbextension to take install only a single nbextension, with an optional destination

### PR DESCRIPTION
**Pull request by jasongrout**
See #7537 for discussion of this.

### CODE REVIEW

**Comment by jasongrout** [comment]
Bonus of having this: --destination now is a command-line argument for the ipython install-nbextension command.

File: IPython/html/nbextensions.py [hunk 1]
```diff
@@ -304,9 +293,9 @@ class NBExtensionApp(BaseIPythonApplication):
     
     Usage
     
-        ipython install-nbextension file [more files, folders, archives or urls]
```
Comment (takluyver) [inline IPython/html/nbextensions.py]: Square brackets usually mean optional arguments. I would just put `path/url`

File: IPython/html/nbextensions.py [hunk 2]
```diff
@@ -304,9 +293,9 @@ class NBExtensionApp(BaseIPythonApplication):
...
+        ipython install-nbextension [file, folder, archive, or url]
     
-    This copies files and/or folders into the IPython nbextensions directory.
```
Comment (takluyver) [inline IPython/html/nbextensions.py]: 'and/or' is no longer relevant, it's just 'or'.

File: IPython/html/nbextensions.py [hunk 3]
```diff
@@ -173,75 +175,61 @@ def install_nbextension(files, overwrite=False, symlink=False, user=False, prefi
...
-                raise ValueError("Cannot symlink from archives")
-            if verbose >= 1:
-                print("extracting %s to %s" % (path, nbext))
-            archive.extractall(nbext)
-            archive.close()
-            continue
-        
-        dest = pjoin(nbext, dest_basename)
-        if overwrite and os.path.exists(dest):
+        else:
```
Comment (takluyver) [inline IPython/html/nbextensions.py]: I don't think this else branch can be reached.

**Comment by jasongrout** [comment]
All very good points.  Fixed!

File: IPython/html/nbextensions.py [hunk 4]
```diff
@@ -173,75 +175,59 @@ def install_nbextension(files, overwrite=False, symlink=False, user=False, prefi
...
-            elif path.endswith('.zip') or _safe_is_tarfile(path):
-                destination = str(uuid.uuid4()) # ignored for archives
-            else:
-                destination = basename(path)
-            _files[destination] = path
-        files = _files
+    if isinstance(path, (list, tuple)):
+        raise TypeError("path must be a string pointing to a single extension to install; call this function multiple times to install multiple extensions")
     
-    for dest_basename,path in (map(cast_unicode_py2, item) for item in files.items()):
```
Comment (takluyver) [inline IPython/html/nbextensions.py]: The previous code called `cast_unicode_py2` on both the path and the destination name; it looks like the new code only calls it on the path.

Comment (jasongrout) [inline IPython/html/nbextensions.py]: Yes, another good point!  Fixed!

**Comment by takluyver** [comment]
I think this looks OK now, but I'll give it a day or two so others can have a look.

**Comment by jasongrout** [comment]
Thanks!

**Pull request merged**